File: old/CodeGenerator/src/main/resources/templates/android/mvc/template.java
package ${packageName};

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

// Model
public class ${projectName}Model {
    private String data;
    
    public ${projectName}Model(String data) {
        this.data = data;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }
}

        // Controller
        public class ${projectName}Controller {
            private ${projectName}Model model;
            private ${projectName}Activity view;

            public ${projectName}Controller(${projectName}Activity view) {
                this.view = view;
                this.model = new ${projectName}Model("");
            }

            public void updateData(String data) {
                model.setData(data);
                view.displayData(data);
            }
        }

        // View
        public class ${projectName}Activity extends AppCompatActivity {
    private ${projectName}Controller controller;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        controller = new ${projectName}Controller(this);
    }

    public void displayData(String data) {
        // Update UI with data
    }
}

File: old/CodeGenerator/src/main/resources/templates/ios/mvc/template.swift
import UIKit

// Model
class ${projectName}Model {
    private var data: String

    init(data: String) {
        self.data = data
    }

    func getData() -> String {
        return data
    }

    func setData(_ data: String) {
        self.data = data
    }
}

// Controller
class ${projectName}Controller {
    private var model: ${projectName}Model
    private weak var view: ${projectName}ViewController?

    init(view: ${projectName}ViewController) {
        self.view = view
        self.model = ${projectName}Model(data: "")
    }

    func updateData(_ data: String) {
        model.setData(data)
        view?.displayData(data)
    }
}

// View
class ${projectName}ViewController: UIViewController {
    private var controller: ${projectName}Controller!

    override func viewDidLoad() {
        super.viewDidLoad()
        controller = ${projectName}Controller(view: self)
    }

    func displayData(_ data: String) {
        // Update UI with data
    }
}
